core/contracts/financial_templates/implementation/TokenFactory.sol

```diff
@@ -0,0 +1,17 @@
+pragma solidity ^0.5.0;
+import "./Token.sol";
+import "../interfaces/TokenInterface.sol";
+
+/**
+ * @notice A factory for creating new tokens and adding the creator as a minter for the newly created tokens.
+ */
+contract TokenFactory {
+    function createToken(string memory tokenName, string memory tokenSymbol, uint8 tokenDecimals)
+        public
```

Comment:
minor question: Any reason why this function can't be external?

Reply:
no i'll add